nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -257,7 +297,52 @@ def _ema_callback(self, trainer: 'pytorch_lightning.Trainer') -> Optional[EMA]:
                 ema_callback = callback
         return ema_callback
 
+    @staticmethod
+    def format_checkpoint_unfinished_marker_path(checkpoint_filepath: Union[Path, str]) -> Path:
+        """
+        Get the path to the unfinished checkpoint marker file.
+        If the file exists, corresponding checkpoint is considered unfinished/incomplete.
+        NOTE: Marker path for the EMA checkpoint part is the same as for the original checkpoint.
+        """
+        marker_filepath = str(uninject_model_parallel_rank(checkpoint_filepath))
+        marker_filepath = marker_filepath.removesuffix(".nemo")
+        marker_filepath = marker_filepath.removesuffix(".ckpt")
+        marker_filepath = marker_filepath.removesuffix("-EMA")
+        return Path(marker_filepath + NeMoModelCheckpoint.UNFINISHED_CHECKPOINT_SUFFIX)
+
+    @staticmethod
+    def is_checkpoint_unfinished(checkpoint_filepath: Union[Path, str]) -> bool:
+        return NeMoModelCheckpoint.format_checkpoint_unfinished_marker_path(checkpoint_filepath).exists()
+
+    @staticmethod
+    def set_checkpoint_unfinished_marker(
+        checkpoint_filepath: Union[Path, str], barrier_before=False, barrier_after=False
+    ) -> None:
+        if barrier_before and torch.distributed.is_initialized():
+            torch.distributed.barrier()
+        if is_global_rank_zero():
+            marker_path = NeMoModelCheckpoint.format_checkpoint_unfinished_marker_path(checkpoint_filepath)
+            marker_path.parent.mkdir(parents=True, exist_ok=True)
+            marker_path.touch()
+        if barrier_after and torch.distributed.is_initialized():
+            torch.distributed.barrier()
+
+    @staticmethod
+    def remove_checkpoint_unfinished_marker(
```

Comment:
Every function should be documented. Especially for core functions that work on save restore

Reply:
Good point. Fixed.